1.8/tutorial/service-discovery.md

```diff
@@ -0,0 +1,65 @@
+  * Let us shh into our cluster and see how this works: `dcos node ssh --master-proxy --leader`,
+  * [Mesos-DNS](https://dcos.io/docs/1.8/usage/service-discovery/mesos-dns/) assigns a Mesos-DNS for every Marathon app.  The naming pattern is  *task.scheduler.mesos* and as the default scheduler for jobs is `marathon`, so the Mesos-DNS name for our redis service is *redis.marathon.mesos*.
+
+  Let us use [dig](https://en.wikipedia.org/wiki/Dig_(command)) to retrieve the *A*ddress record: `dig redis.marathon.mesos`.
+     The answer should contain something like the following response:
+
+  ~~~~
+  ;; ANSWER SECTION:
+  redis.marathon.mesos. 60  IN  A 10.0.0.43
+  ~~~~
+
+  The response tells us that the host for the 'redis.marathon.mesos'' is 10.0.0.43.
+
+  The *A*ddress record only contains information abbout the host, which is not sufficient to connect to the service as we also need to know the port. The Service locator (SRV) DNS record includes port information as well: `dig srv _redis._tcp.marathon.mesos`.
+  The answer should look similar to this response:
+
+  ~~~~
+  ;; ANSWER SECTION:
+  _redis._tcp.marathon.mesos. 60  IN  SRV 0 0 30585 redis-1y1hj-s1.marathon.mesos.
+
+  ;; ADDITIONAL SECTION:
+  redis-1y1hj-s1.marathon.mesos. 60 IN  A 10.0.0.43
+  ~~~~
+
+   So we now know that our redis app is running on 10.0.0.43:30585.
+
+    * [Named Vips](https://dcos.io/docs/1.8/usage/service-discovery/load-balancing-vips/) allow you to assign name/port pairs to your apps.
+    For example the package defintion for our redis service contains the following label:
+
+    ~~~
+    "VIP_0": "/redis:6379"
+    ~~~
+
+    The full name is then generated using the following schema:
+    *vip-name.scheduler.l4lb.thisdcos.directory:vip-port*. Hence, we can use *redis.marathon.l4lb.thisdcos.directory:6379 to reach our redis service from within the cluster.
+
+# Outcome
+We explored how service-discovery enables us to find our tasks within the DC/OS cluster.
+
+# Deep Dive
+So what are the differences between Mesos-DNS and Named VIPs?
+Mesos-DNS is a rather simple solution to find applications inside the cluster and DNS is supported by many applications.
+But DNS entries have several disadvantages, including:
+
+  * DNS-Caching: Applications might cache DNS entries for efficiency and  hence not use an updated address information (e.g., after a task failure).
+  * SRV Records: We have seen that in order to retrieve information about the allocated ports we need to utilize SRV DNS records. Even though DNS A records are commonly understood by applications, not all applications can deal with SRV records.
+
+Named VIPs actually load-balance the IP address port pair, and hence also redirects to the current instance in cases where the applications cached the IP address. Furthermore, they allow you to select a port as well. Hence, you should consider Named VIPs are the default option for service discovery in DC/OS.
```

Comment:
Suggest: `Named VIPs load balance the IP address/port pair and therefore also redirects the current instance when applications have cached the IP address. They also allow you to select a port. Because Named VIPs offer these advantages over Mesos-DNS, we suggest using Named VIPs as the default service discovery method in DC/OS.`